src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -478,8 +480,8 @@ async Task<bool> NextResult(bool async, bool isConsuming = false, CancellationTo
                 {
                     // Statement did not generate a resultset (e.g. INSERT)
                     // Read and process its completion message and move on to the next statement
-
-                    msg = await ReadMessage(async);
+                    // No need to read sequentially as it's not a DataRow
```

Comment:
We can rename NpgsqlDataReader.ReadMessage to ReadDataRow and add an assertion to make sure it only reads data rows out.. This way any other message must go directly through connector.ReadMessage

Reply:
We can't always guarantee that.

https://github.com/npgsql/npgsql/blob/fc50c7e5b313b6b58de9f549b056d8e24a1e2290/src/Npgsql/NpgsqlDataReader.cs#L514-L528

For example, here we can get either `DataRow`, or `CommandComplete`.

Reply:
Right, makes sense.